common/kvstore/leveldb/leveldb_store.go

```diff
@@ -16,27 +18,40 @@ var _ kvstore.Store[[]byte] = &levelDBStore{}
 
 // levelDBStore implements kvstore.Store interfaces with levelDB as the backend engine.
 type levelDBStore struct {
-	db   *leveldb.DB
-	path string
-
-	logger logging.Logger
-
+	db       *leveldb.DB
+	path     string
 	shutdown bool
+	logger   logging.Logger
+	metrics  *MetricsCollector
 }
 
 // NewStore returns a new levelDBStore built using LevelDB.
 func NewStore(logger logging.Logger, path string) (kvstore.Store[[]byte], error) {
+	return NewStoreWithMetrics(logger, path, nil)
+}
+
+// NewStoreWithMetrics returns a new levelDBStore built using LevelDB with metrics collection.
+// If reg is nil, metrics will not be collected.
+func NewStoreWithMetrics(logger logging.Logger, path string, reg *prometheus.Registry) (kvstore.Store[[]byte], error) {
```

Comment:
A personal preference, feel free to push back if you disagree. I'd be ok only having one constructor. Not too burdensome passing in `nil` when we don't care about metrics. The only place where we probably won't be using metrics is going to be in testing, I'd presume.

Reply:
The question may be who are "we", ie. whether it makes sense for all validators to store and monitor these metrics by default.
If it's small amount of extra resource, should be fine (even if they never look at it, it may help us/developers troubleshooting when needed); but if it's non-trivial, we'd better flag guarded it.

Reply:
Single constructor seems reasonable. I can add a flag enabled by default, the overhead of the metrics calculation is not significant.

Reply:
Ok, I saw all but one are gauge which is cheaper type of metrics (if not with high cardinality).

Reply:
`compaction_duration_seconds` is a histogram. That one can potentially have a lot of cardinality if there's multiple levels (large DB).

Reply:
https://github.com/Layr-Labs/eigenda/pull/1313/commits/5ccacd859ce3f9835c269d74e0fa19efba36d77d